src/MPAGSCipher/CaesarCipher.cpp

```diff
@@ -0,0 +1,90 @@
+#include "CaesarCipher.hpp"
+#include "CipherMode.hpp"
+
+#include <string>
+#include <iostream>
+
+// class CaesarCipher {
+//     public:
+//     explicit CaesarCipher(const std::size_t& key);
+
+//     explicit CaesarCipher(const std::string& key);
+
+//     std::string applyCipher (const std::string& inputText, const CipherMode encrypt) const;
+
+//     private:
+
+//     std::size_t key_ ;
+//     const std::vector<char> alphabet_;
+
+
+// };
+
+CaesarCipher::CaesarCipher(const std::size_t& key)
+:key_{0},alphabet_{'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'}
+{
+    key_ = key;
+}
```

Comment:
Given the suggested changes in the corresponding header file, this can be simplified:
```suggestion
CaesarCipher::CaesarCipher(const std::size_t& key)
:key_{key % alphabet_.size()}
{
}
```